test/tftp/tftp_server_test.rb

```diff
@@ -0,0 +1,110 @@
+require 'test_helper'
+require 'tftp/server'
+require 'tftp/tftp_plugin'
+
+module TftpGenericServerSuite
+  def setup
+    @rootdir = "/some/root"
+    @mac = "aa:bb:cc:dd:ee:ff"
+    @content = "file content"
+    FileUtils.stubs(:mkdir_p).returns(true)
+    Proxy::TFTP::Plugin.settings.stubs(:tftproot).returns(@rootdir)
+    setup_paths
+  end
+
+  def pxe_config_files
+    @pxe_config_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def pxe_default_files
+    @pxe_default_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def test_set
+    pxe_config_files.each do |file|
+      @subject.stubs(:write_file).with(file, @content)
+    end
+    @subject.expects(:write_file).with(pxe_config_files.first, @content).once
```

Comment:
Mocha supports mocking of repeated calls with different parameter values, including verification of the order of calls:
 :
```
  class A
    def a(arg); end
    def b
      a(:first)
      a(:second)
    end
  end

  def test_multiple_mocks_disregarding_call_order
    under_test = A.new

    under_test.expects(:a).with(:first)
    under_test.expects(:a).with(:second).once

    under_test.b
  end

  def test_multiple_mocks_in_order
    under_test = A.new
    call_sequence = sequence('test_sequence')

    under_test.expects(:a).with(:first, :first_1).in_sequence(call_sequence)
    under_test.expects(:a).with(:second, :second_1).in_sequence(call_sequence)

    under_test.b
  end
```